build.gradle

```diff
@@ -110,6 +119,12 @@ test {
         exceptionFormat = 'full'
         events "started", "passed", "skipped", "failed"
     }
+    retry {
+        failOnPassedAfterRetry = false
+        maxFailures = 5
+        maxRetries = 5
+    }
+    maxParallelForks = Runtime.runtime.availableProcessors()
```

Comment:
I've also enabled running tests in parallel which can greatly reduce test time for systems with many cores, this change doesn't appear to have much effect on how often test failures occur.